test/sharness/t0250-files-api.sh

```diff
@@ -0,0 +1,219 @@
+#!/bin/sh
+#
+# Copyright (c) 2015 Jeromy Johnson
+# MIT Licensed; see the LICENSE file in this repository.
+#
+
+test_description="test the unix files api"
```

Comment:
great! this is a good basic test. also needs:

- [ ] opts on root '/'
  - [ ] test cannot mkdir `/`
  - [ ] test cannot `rm /` (or, if done, an empty `/` is still there at the end).
- [ ] test `--offset` and `--count` with large files, both small and large numbers.
- [ ] for read and write:
  - [ ] for --offset
    - [ ] test negative
    - [ ] test 0
    - [ ] test last byte
    - [ ] test > last byte
  - [ ] for --count
    - [ ] test negative
    - [ ] test 0
    - [ ] length
    - [ ] > length
- [ ] test cannot `write` to dir paths: 
  - [ ] to a path like  `/cats/foo/existingdir` which might be a file, but is an existing dir
  - [ ] to a path like `/cats/foo/nonexistant/` which is nonexistant, but ends in `/`